install/macos/build-app.py

```diff
@@ -37,96 +59,104 @@ def codesign(path, entitlements, identity):
     )
 
 
+def sign_app_bundle(build_path, dist_path, app_path):
+    """
+    Signs the app bundle stored in dist/Dangerzone.app,
+    producing a Dangerzone.dmg file
+    """
+    print(f"○ Signing app bundle in {app_path}")
+
+    # Detect if create-dmg is installed
+    if not os.path.exists(app_path):
+        print(f"ERROR: Dangerzone.app not found in {app_path}.")
+        exit(1)
+
+    dmg_path = os.path.join(dist_path, "Dangerzone.dmg")
+    icon_path = os.path.join(root, "install", "macos", "dangerzone.icns")
+
+    print("○ Code signing app bundle")
+    identity_name_application = (
+        # FIXME: Update this line with the proper developer ID, once we know what
+        # that is.
+        "Developer ID Application: FREEDOM OF THE PRESS FOUNDATION, 501(c)(3)"
+    )
+    entitlements_plist_path = os.path.join(root, "install/macos/entitlements.plist")
+
+    for path in itertools.chain(
+        glob.glob(f"{app_path}/**/*.so", recursive=True),
+        glob.glob(f"{app_path}/**/*.dylib", recursive=True),
+        glob.glob(f"{app_path}/**/Python3", recursive=True),
+        [app_path],
+    ):
+        codesign(path, entitlements_plist_path, identity_name_application)
+    print(f"○ Signed app bundle: {app_path}")
+
+    # Detect if create-dmg is installed
+    if not os.path.exists("/usr/local/bin/create-dmg"):
+        print("create-dmg is not installed, skipping creating a DMG")
+        return
+
+    print("○ Creating DMG")
+    run(
+        [
+            "create-dmg",
+            "--volname",
+            "Dangerzone",
+            "--volicon",
+            icon_path,
+            "--window-size",
+            "400",
+            "200",
+            "--icon-size",
+            "100",
+            "--icon",
+            "Dangerzone.app",
+            "100",
+            "70",
+            "--hide-extension",
+            "Dangerzone.app",
+            "--app-drop-link",
+            "300",
+            "70",
+            dmg_path,
+            app_path,
+            "--identity",
+            identity_name_application,
+        ]
+    )
+    print(f"○ Finished building DMG: {dmg_path}")
```

Comment:
Minor, but since this action is now separated from the building of the DMG, it would probably make sense to rename it to "Finished signing DMG".

Reply:
Actually, as far as I undestand, the DMG is only produced when the signing is enabled ([it was the case before](https://github.com/freedomofpress/dangerzone/pull/384/files#diff-9e3e8e5c0a05684d57b4ae2d4184a72ffc7ae648ee9627faa4a91a4024e728d7L74)). So I think we can keep it as is, but if you prefer I can also say that it's finished building and signing the DMG.

How would you prefer?

Reply:
Oh, thanks for the explanation. Let's keep it then.